### ISSUE TITLE
allow setting metadata

### ISSUE THREAD

**Issue by butonic** [cernbox/cs3apis#7]
We should allow sending MD Metadata with
- [ ] FS.CreateDir and
- [ ] FS.Upload to send along the original ctime and mtime as well as extended attributes.

Furthermore, we need to be able to manipulate metadata, in order to
- [ ] implement touch()
- [ ] set extended attributes

**Comment by moscicki** [comment]
@butonic: correct. we are cleaning all these things up properly on the review branch.

**Comment by labkode** [comment]
@butonic the opaque field helps in this case, for example here:
https://github.com/cernbox/cs3apis/blob/review/cs3/storageprovider/v0alpha/storageprovider.proto#L157

If we know that there will be logic used by many implementations then those use-cases can be promoted to non-opaque fields.

**Comment by moscicki** [comment]
We should add attributes to ExtraMetadata wherever it makes sense and does not depend on particular storage backend:

https://github.com/cernbox/cs3apis/blob/review/cs3/storageprovider/v0alpha/resources.proto#L138

I would avoid creating too many hidden stuff in Opaque because it will be breaking the "standard".

**Comment by labkode** [comment]
> We should add attributes to ExtraMetadata wherever it makes sense and does not depend on particular storage backend:
> 
> https://github.com/cernbox/cs3apis/blob/review/cs3/storageprovider/v0alpha/resources.proto#L138
> 
> I would avoid creating too many hidden stuff in Opaque because it will be breaking the "standard".

If many storage back ends support it
